feg/gateway/services/n7_n40_proxy/n7/n7.go

```diff
@@ -39,3 +50,509 @@ func NewN7Client(cfg *PCFConfig) (*n7_sbi.ClientWithResponses, error) {
 	)
 	return client, err
 }
+
```

Comment:
see how this is done at other services [here](http://github.com/magma/magma/blob/master/feg/gateway/services/session_proxy/credit_control/gx/model_conversion.go#L32-L32)

model_conversion files have different kind of functions that comverst from proto to diameter message and reverse
- [This](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Amagma%2Fmagma.git&branch=master&file=feg%2Fgateway%2Fservices%2Fsession_proxy%2Fcredit_control%2Fgx%2Fmodel_conversion.go&editor=JetBrains&version=v1.2.1&utm_product_name=IntelliJ+IDEA&utm_product_version=2021.2.2&start_row=84&start_col=1&end_row=84&end_col=1) converts from diameter structure to proto
- [This](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Amagma%2Fmagma.git&branch=master&file=feg%2Fgateway%2Fservices%2Fsession_proxy%2Fcredit_control%2Fgx%2Fmodel_conversion.go&editor=JetBrains&version=v1.2.1&utm_product_name=IntelliJ+IDEA&utm_product_version=2021.2.2&start_row=66&start_col=31&end_row=66&end_col=31) converts from proto to diameter struct

Note how in this case we do use pointers receivers to extend `RuleDefinition` or `QosRequestInfo`.  In cases like [this](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Amagma%2Fmagma.git&branch=master&file=feg%2Fgateway%2Fservices%2Fsession_proxy%2Fcredit_control%2Fgx%2Fmodel_conversion.go&editor=JetBrains&version=v1.2.1&utm_product_name=IntelliJ+IDEA&utm_product_version=2021.2.2&start_row=100&start_col=11&end_row=100&end_col=11), (for strings types) we just indicate the action we do on the function (that function convert from diameter type to proto type)

Reply:
Don't think we can use pointer receivers for structs defined in another package. Will get this error - "invalid receiver *protos.CreateSessionRequest (type not defined in this package)"

Reply:
You are right. In case Gx.Gy the base structs are part of diameter but messages are part of definition.go that belongs to gx.

Then you can ignore all the comments related to that. But please, find a consistent way to label all them. You will have functions that convert from proto to sbi and from sbi to proto